test/e2e/multihoming.go

```diff
@@ -627,254 +628,312 @@ var _ = Describe("Multi Homing", func() {
+					metav1.DeleteOptions{PropagationPolicy: &blockUntilEverythingIsGone},
+				)).To(Succeed())
+				Eventually(func() bool {
+					_, err := cs.CoreV1().Namespaces().Get(context.Background(), extraNamespace.Name, metav1.GetOptions{})
+					nsPods, podCatchErr := cs.CoreV1().Pods(extraNamespace.Name).List(context.Background(), metav1.ListOptions{})
+					return podCatchErr == nil && errors.IsNotFound(err) && len(nsPods.Items) == 0
+				}, 2*time.Minute, 5*time.Second).Should(BeTrue())
+			})
 
-				By("asserting the server pod reaches the `Ready` state")
-				Eventually(func() v1.PodPhase {
-					updatedPod, err := cs.CoreV1().Pods(serverPodConfig.namespace).Get(context.Background(), serverPod.GetName(), metav1.GetOptions{})
-					if err != nil {
-						return v1.PodFailed
+			table.DescribeTable(
+				"multi-network policies configure traffic allow lists",
+				func(netConfig networkAttachmentConfig, allowedClientPodConfig podConfiguration, blockedClientPodConfig podConfiguration, serverPodConfig podConfiguration, policy *mnpapi.MultiNetworkPolicy) {
+					if blockedClientPodConfig.namespace == "" {
+						blockedClientPodConfig.namespace = f.Namespace.Name
+					} else {
+						blockedClientPodConfig.namespace = extraNamespace.Name
 					}
-					return updatedPod.Status.Phase
-				}, 2*time.Minute, 6*time.Second).Should(Equal(v1.PodRunning))
-
-				By("instantiating the *allowed-client* pod")
-				allowedClient, err := cs.CoreV1().Pods(allowedClientPodConfig.namespace).Create(
-					context.Background(),
-					generatePodSpec(allowedClientPodConfig),
-					metav1.CreateOptions{},
-				)
-				Expect(err).NotTo(HaveOccurred())
-
-				By("asserting the *allowed-client* pod reaches the `Ready` state")
-				Eventually(func() v1.PodPhase {
-					updatedPod, err := cs.CoreV1().Pods(allowedClientPodConfig.namespace).Get(context.Background(), allowedClient.GetName(), metav1.GetOptions{})
-					if err != nil {
-						return v1.PodFailed
+					if allowedClientPodConfig.namespace == "" {
+						allowedClientPodConfig.namespace = f.Namespace.Name
+					} else {
+						allowedClientPodConfig.namespace = extraNamespace.Name
 					}
-					return updatedPod.Status.Phase
-				}, 2*time.Minute, 6*time.Second).Should(Equal(v1.PodRunning))
-
-				By("instantiating the *blocked-client* pod")
-				blockedClient, err := cs.CoreV1().Pods(blockedClientPodConfig.namespace).Create(
-					context.Background(),
-					generatePodSpec(blockedClientPodConfig),
-					metav1.CreateOptions{},
-				)
-				Expect(err).NotTo(HaveOccurred())
-
-				By("asserting the *blocked-client* pod reaches the `Ready` state")
-				Eventually(func() v1.PodPhase {
-					updatedPod, err := cs.CoreV1().Pods(blockedClientPodConfig.namespace).Get(context.Background(), blockedClient.GetName(), metav1.GetOptions{})
-					if err != nil {
-						return v1.PodFailed
+					serverPodConfig.namespace = f.Namespace.Name
+
+					for _, ns := range []v1.Namespace{*f.Namespace, *extraNamespace} {
```

Comment:
Would it have been an option to create a single NAD and make the pods refer to it including the namespace in the reference? Like for example in the default namespace. Not a biggie, but I would also probably add a comment stating how this is done for convinience but that both NADs refer to the same network or something in that vein.

Reply:
Resolving.